Framework/Algorithms/src/ReflectometryReductionOneAuto3.cpp

```diff
@@ -0,0 +1,1126 @@
+      "When correcting detector positions, this determines whether detectors"
+      "should be shifted vertically or rotated around the sample position.",
+      Direction::Input);
+  setPropertySettings("DetectorCorrectionType",
+                      make_unique<Kernel::EnabledWhenProperty>(
+                          "CorrectDetectors", IS_EQUAL_TO, "1"));
+
+  // Wavelength limits
+  declareProperty("WavelengthMin", Mantid::EMPTY_DBL(),
+                  "Wavelength Min in angstroms", Direction::Input);
+  declareProperty("WavelengthMax", Mantid::EMPTY_DBL(),
+                  "Wavelength Max in angstroms", Direction::Input);
+
+  // Monitor properties
+  initMonitorProperties();
+
+  // Init properties for transmission normalization
+  initTransmissionProperties();
+
+  // Init properties for algorithmic corrections
+  initAlgorithmicProperties(true);
+
+  // Momentum transfer properties
+  initMomentumTransferProperties();
+
+  // Polarization correction
+  declareProperty(make_unique<PropertyWithValue<bool>>("PolarizationAnalysis",
+                                                       false, Direction::Input),
+                  "Apply polarization corrections");
+
+  // Flood correction
+  std::vector<std::string> propOptions = {"Workspace", "ParameterFile"};
+  declareProperty("FloodCorrection", "Workspace",
+                  boost::make_shared<StringListValidator>(propOptions),
+                  "The way to apply flood correction: "
+                  "Workspace - use FloodWorkspace property to get the flood "
+                  "workspace, ParameterFile - use parameters in the parameter "
+                  "file to construct and apply flood correction workspace.");
+  declareProperty(
+      make_unique<WorkspaceProperty<MatrixWorkspace>>(
+          "FloodWorkspace", "", Direction::Input, PropertyMode::Optional),
+      "A flood workspace to apply; if empty and FloodCorrection is "
+      "'Workspace' then no correction is applied.");
+
+  // Init properties for diagnostics
+  initDebugProperties();
+
+  // Output workspace in Q
+  declareProperty(make_unique<WorkspaceProperty<MatrixWorkspace>>(
+                      "OutputWorkspaceBinned", "", Direction::Output,
+                      PropertyMode::Optional),
+                  "Output workspace in Q (rebinned workspace)");
+
+  // Output workspace in Q (unbinned)
+  declareProperty(
+      make_unique<WorkspaceProperty<MatrixWorkspace>>(
+          "OutputWorkspace", "", Direction::Output, PropertyMode::Optional),
+      "Output workspace in Q (native binning)");
+
+  // Output workspace in wavelength
+  declareProperty(make_unique<WorkspaceProperty<MatrixWorkspace>>(
+                      "OutputWorkspaceWavelength", "", Direction::Output,
+                      PropertyMode::Optional),
+                  "Output workspace in wavelength");
+}
+
+/** Execute the algorithm.
+ */
+void ReflectometryReductionOneAuto3::exec() {
+
```

Comment:
Can these excess newlines at the start of functions be removed?

They hurt readability.